File: AppGenerateFiles/dbo/FrontModel/Periodo_lectivos.js
//@ts-check
import { EntityClass } from "../../../WDevCore/WModules/EntityClass.js";
import { WAjaxTools, BasicStates } from "../../../WDevCore/WModules/WComponentsTools.js";
//@ts-ignore
import { ModelProperty } from "../../../WDevCore/WModules/CommonModel.js";
import { Estudiante_clases }  from './Estudiante_clases.js'
class Periodo_lectivos extends EntityClass {
   /** @param {Partial<Periodo_lectivos>} [props] */
   constructor(props) {
       super(props, 'EntityDbo');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   /**@type {Number}*/ Id;
   /**@type {String}*/ Nombre;
   /**@type {String}*/ Nombre_corto;
   /**@type {Date}*/ Inicio;
   /**@type {Date}*/ Fin;
   /**@type {String}*/ Observaciones;
   /**@type {String}*/ Config;
   /**@type {Date}*/ Created_at;
   /**@type {Date}*/ Updated_at;
   /**@type {Boolean}*/ Abierto;
   /**@type {Boolean}*/ Oculto;
   /**@type {Array<Estudiante_clases>} OneToMany*/ Estudiante_clases;
}
export { Periodo_lectivos }

File: AppGenerateFiles/dbo/Model/Periodo_lectivos.cs
using CAPA_DATOS;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace DataBaseModel {
   public class Periodo_lectivos : EntityClass {
       [PrimaryKey(Identity = false)]
       public int? Id { get; set; }
       public string? Nombre { get; set; }
       public string? Nombre_corto { get; set; }
       public DateTime? Inicio { get; set; }
       public DateTime? Fin { get; set; }
       public string? Observaciones { get; set; }
       public string? Config { get; set; }
       public DateTime? Created_at { get; set; }
       public DateTime? Updated_at { get; set; }
       public bool? Abierto { get; set; }
       public bool? Oculto { get; set; }
       [OneToMany(TableName = "Estudiante_clases", KeyColumn = "Id", ForeignKeyColumn = "Periodo_lectivo_id")]
       public List<Estudiante_clases>? Estudiante_clases { get; set; }
   }
}
